webdriver/tests/bidi/browsing_context/locate_nodes/invalid.py

```diff
@@ -0,0 +1,171 @@
+async def test_params_locator_invalid_type(bidi_session, inline, top_context, value):
+    url = inline("""<div>foo<span><strong>bar</strong></span><span>BAR</span>baz</div>""")
+    await bidi_session.browsing_context.navigate(
+        context=top_context["context"], url=url, wait="complete"
+    )
+
+    with pytest.raises(error.InvalidArgumentException):
+        await bidi_session.browsing_context.locate_nodes(
+            context=top_context["context"], locator={ "type": value, "value": "div" }
+        )
+
+
+async def test_params_locator_invalid_value(bidi_session, inline, top_context):
+    url = inline("""<div>foo<span><strong>bar</strong></span><span>BAR</span>baz</div>""")
+    await bidi_session.browsing_context.navigate(
+        context=top_context["context"], url=url, wait="complete"
+    )
+
+    with pytest.raises(error.InvalidArgumentException):
+        await bidi_session.browsing_context.locate_nodes(
+            context=top_context["context"], locator={ "type": "invalid", "value": "div" }
+        )
+
+
+@pytest.mark.parametrize("value", [False, "string", {}, []])
+async def test_params_max_node_count_invalid_type(bidi_session, inline, top_context, value):
+    url = inline("""<div>foo<span><strong>bar</strong></span><span>BAR</span>baz</div>""")
+    await bidi_session.browsing_context.navigate(
+        context=top_context["context"], url=url, wait="complete"
+    )
+
+    with pytest.raises(error.InvalidArgumentException):
+        await bidi_session.browsing_context.locate_nodes(
+            context=top_context["context"],
+            locator={ "type": "css", "value": "div" },
+            max_node_count=value
+        )
+
+
+async def test_params_max_node_count_invalid_value(bidi_session, inline, top_context):
+    url = inline("""<div>foo<span><strong>bar</strong></span><span>BAR</span>baz</div>""")
+    await bidi_session.browsing_context.navigate(
+        context=top_context["context"], url=url, wait="complete"
+    )
+
+    with pytest.raises(error.InvalidArgumentException):
+        await bidi_session.browsing_context.locate_nodes(
+            context=top_context["context"],
+            locator={ "type": "invalid", "value": "div" },
+            max_node_count=0
+        )
+
+
+@pytest.mark.parametrize("value", [False, "string", 42, {}])
+async def test_params_start_nodes_invalid_type(bidi_session, inline, top_context, value):
+    url = inline("""<div>foo<span><strong>bar</strong></span><span>BAR</span>baz</div>""")
+    await bidi_session.browsing_context.navigate(
+        context=top_context["context"], url=url, wait="complete"
+    )
+
+    with pytest.raises(error.InvalidArgumentException):
+        await bidi_session.browsing_context.locate_nodes(
+            context=top_context["context"],
+            locator={ "type": "css", "value": "div" },
+            start_nodes=value
+        )
+
+
+async def test_params_start_nodes_invalid_value(bidi_session, inline, top_context):
```

Comment:
I would call this `empty list` to better reflect what we are testing here. Note that we should have at least a test with an invalid type/value for the shared reference's id. Optional arguments we probably won't have to test here because it pretty well covered by script evaluation.

```suggestion
async def test_params_start_nodes_empty_list(bidi_session, inline, top_context):
```